### PR TITLE
Update to elastic-agent-client with chunking support.

### PR DESCRIPTION
**Pull request by blakerouse**
## Proposed commit message

Update to elastic-agent-client with chunking support.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- ~~[ ] I have made corresponding changes to the documentation~~
- ~~[ ] I have made corresponding change to the default configuration files~~
- ~~[ ] I have added tests that prove my fix is effective or that my feature works~~
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Author's Checklist

- [x] Enables chunking support when communicating with the Elastic Agent.

## How to test this PR locally

See https://github.com/elastic/elastic-agent/pull/3884 for how to test.

## Related issues

- Relates https://github.com/elastic/elastic-agent/issues/2460

### CODE REVIEW

**Comment by blakerouse** [comment]
go.mod replace has been removed, now that v7.6.0 of elastic-agent-client has been released

File: x-pack/libbeat/management/simple_input_config_test.go [hunk 1]
```diff
@@ -133,7 +133,7 @@ func TestSimpleInputConfig(t *testing.T) {
 		fmt.Sprintf(":%d", server.Port),
 		"",
 		client.VersionInfo{},
-		grpc.WithTransportCredentials(insecure.NewCredentials()))
+		client.WithGRPCDialOptions(grpc.WithTransportCredentials(insecure.NewCredentials())))
```
Comment (cmacknz) [inline x-pack/libbeat/management/simple_input_config_test.go:136]: Can the test in here be modified to verify that chunking works as expected?

Comment (blakerouse) [reply x-pack/libbeat/management/simple_input_config_test.go:136]: Doesn't make sense to test a feature of an external library. The library handles all the chunking and ensures that its on, and that library is fully tested.

Comment (cmacknz) [reply x-pack/libbeat/management/simple_input_config_test.go:136]: In this case where there are essentially no changes besides the dependency this seems reasonable.

I am a bit worried that nothing is actually testing that this works with a real Beat anywhere though. If we can force an existing integration test in the agent repository to use chunking for everything that's probably a better way to test this.

**Review by cmacknz — APPROVED** [review]

**Pull request merged**